vlog4j-examples/src/main/java/org/semanticweb/vlog4j/examples/core/AddDataFromCsvFile.java

```diff
@@ -0,0 +1,167 @@
+package org.semanticweb.vlog4j.examples.core;
+
+/*-
+ * #%L
+ * VLog4j Examples
+ * %%
+ * Copyright (C) 2018 VLog4j Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeAtom;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConjunction;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConstant;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makePredicate;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeRule;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeVariable;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.semanticweb.vlog4j.core.model.api.Atom;
+import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Variable;
+import org.semanticweb.vlog4j.core.reasoner.DataSource;
+import org.semanticweb.vlog4j.core.reasoner.Reasoner;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.EdbIdbSeparationException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.IncompatiblePredicateArityException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.ReasonerStateException;
+import org.semanticweb.vlog4j.core.reasoner.implementation.CsvFileDataSource;
+import org.semanticweb.vlog4j.examples.ExamplesUtils;
+
+/**
+ * This example shows how facts can be imported from {@code .csv.gz} files, and
+ * also how query answers can be exported to {@code .csv} files. <i>Note that
+ * you can also import from {@code .csv} files.</i>
+ *
+ * For importing, a {@link CsvFileDataSource} that contains a path to the
+ * corresponding {@code .csv.gz} file must be created. A {@code .csv} file
+ * contains facts over exactly one predicate in the CSV format, while a
+ * {@code .csv.gz} file is the gzipped version of such a {@code .csv} file. For
+ * exporting, a path to the output {@code .csv} file must be specified.
```

Comment:
Before "For exporting..." I would add, in a new paragraph: "Also, facts resulting from answering a query can be exported into .csv files. "

Reply:
This information is already given in the first paragraph of this example. _"This example shows [...] how query answers can be exported to .csv files."_ I can reformat the Javadoc a bit, but I wouldn't change any content.